src/gwt/src/org/rstudio/studio/client/application/Application.java

```diff
@@ -206,7 +230,15 @@ public void run()
                dismissLoadingProgress.execute();
             }
 
+            // set session info
             session_.setSessionInfo(sessionInfo);
+            
+            // initialize application hooks
+            if (sessionInfo.isAutomationAgent())
+            {
+               ApplicationAutomationHooks appHooks = pApplicationHooks_.get();
+               appHooks.initialize();
```

Comment:
The intention here is to export some extra Javascript callbacks to make automation testing easier, but only do it specifically for RStudio instances launched as automation agents.